Formula/wine.rb

```diff
@@ -64,53 +51,421 @@ def require_universal_deps?
+    sha256 "137f13a7eb93ce72e32ccd7cebdab6874f8cf7ddf31d3a455a68e016ecd9e4e6"
+  end
+
+  resource "libgphoto2" do
+    url "https://downloads.sourceforge.net/project/gphoto/libgphoto/2.5.12/libgphoto2-2.5.12.tar.bz2"
+    mirror "https://fossies.org/linux/privat/libgphoto2-2.5.12.tar.bz2"
+    sha256 "b9bb28990fde45ac385e4851a07dbad2e1250404b535b0a3a3b898bb431e4e2e"
+  end
+
+  resource "net-snmp" do
+    url "https://downloads.sourceforge.net/project/net-snmp/net-snmp/5.7.3/net-snmp-5.7.3.tar.gz"
+    sha256 "12ef89613c7707dc96d13335f153c1921efc9d61d3708ef09f3fc4a7014fb4f0"
+  end
+
+  resource "sane-backends" do
+    url "https://mirrors.kernel.org/debian/pool/main/s/sane-backends/sane-backends_1.0.25.orig.tar.gz"
+    mirror "https://fossies.org/linux/misc/sane-backends-1.0.25.tar.gz"
+    sha256 "a4d7ba8d62b2dea702ce76be85699940992daf3f44823ddc128812da33dc6e2c"
+  end
+
+  # Fixes some missing headers missing error. Reported upstream
+  # https://lists.alioth.debian.org/pipermail/sane-devel/2015-October/033972.html
+  resource "sane-backends-patch" do
+    url "https://raw.githubusercontent.com/Homebrew/formula-patches/6dd7790c/sane-backends/1.0.25-missing-types.patch"
+    sha256 "f1cda7914e95df80b7c2c5f796e5db43896f90a0a9679fbc6c1460af66bdbb93"
   end
 
   fails_with :clang do
     build 425
     cause "Clang prior to Xcode 5 miscompiles some parts of wine"
   end
 
-  # These libraries are not specified as dependencies, or not built as 32-bit:
-  # configure: libv4l, gstreamer-0.10, libcapi20, libgsm
+  def openssl_arch_args
+    {
+      :x86_64 => %w[darwin64-x86_64-cc enable-ec_nistp_64_gcc_128],
+      :i386 => %w[darwin-i386-cc],
+    }
+  end
+
+  # Store and restore some of our environment
+  def save_env
+    saved_cflags = ENV["CFLAGS"]
+    saved_ldflags = ENV["LDFLAGS"]
+    saved_homebrew_archflags = ENV["HOMEBREW_ARCHFLAGS"]
+    saved_homebrew_cccfg = ENV["HOMEBREW_CCCFG"]
+    saved_makeflags = ENV["MAKEFLAGS"]
+    saved_homebrew_optflags = ENV["HOMEBREW_OPTFLAGS"]
+    begin
+      yield
+    ensure
+      ENV["CFLAGS"] = saved_cflags
+      ENV["LDFLAGS"] = saved_ldflags
+      ENV["HOMEBREW_ARCHFLAGS"] = saved_homebrew_archflags
+      ENV["HOMEBREW_CCCFG"] = saved_homebrew_cccfg
+      ENV["MAKEFLAGS"] = saved_makeflags
+      ENV["HOMEBREW_OPTFLAGS"] = saved_homebrew_optflags
+    end
+  end
 
   def install
+    ENV.prepend_create_path "PATH", "#{libexec}/bin"
+    ENV.prepend_create_path "PKG_CONFIG_PATH", "#{libexec}/lib/pkgconfig"
+
+    resource("openssl").stage do
+      save_env do
+        ENV.deparallelize
+        ENV.permit_arch_flags
+
+        # OpenSSL will prefer the PERL environment variable if set over $PATH
+        # which can cause some odd edge cases & isn't intended. Unset for safety,
+        # along with perl modules in PERL5LIB.
+        ENV.delete("PERL")
+        ENV.delete("PERL5LIB")
+
+        # Load zlib from an explicit path instead of relying on dyld's fallback
```

Comment:
Unless `wine` is using TLS compression for some reason, you could probably just configure this vendored `openssl` with `no-zlib` instead of carrying a duplicate `inreplace` until the end of days.